workflows/flows/import_genomes_flow.py

```diff
@@ -0,0 +1,355 @@
+        catalogue_id=catalogue_id,
+        defaults={
+            "version": options["catalogue_version"],
+            "name": f"{options['catalogue_name']} v{options['catalogue_version']}",
+            "biome": biome,
+            "result_directory": options["catalogue_dir"],
+            "ftp_url": "http://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/",
+            "pipeline_version_tag": options["pipeline_version"],
+            "catalogue_biome_label": options["catalogue_biome_label"],
+            "catalogue_type": options["catalogue_type"],
+        },
+    )
+    return catalogue
+
+
+def gather_genome_dirs(catalogue_dir, catalogue_type):
+    genome_dirs = find_genome_results(catalogue_dir)
+
+    sanity_check_map = {
+        "eukaryotes": sanity_check_genome_output_euks,
+        "prokaryotes": sanity_check_genome_output_proks,
+    }
+    sanity_check = sanity_check_map.get(catalogue_type)
+    if sanity_check:
+        for d in genome_dirs:
+            sanity_check(d)
+
+    sanity_check_catalogue_dir(catalogue_dir)
+    return genome_dirs
+
+
+@task
+def process_genome_dir(catalogue, genome_dir):
+    accession = apparent_accession_of_genome_dir(genome_dir)
+    logger = get_run_logger()
+    logger.info(f"Processing genome: {accession}")
+
+    genome_data = read_json(os.path.join(genome_dir, f"{accession}.json"))
+    has_pangenome = "pangenome" in genome_data
+
+    path = Biome.lineage_to_path(genome_data["gold_biome"])
+
+    genome_data["catalogue"] = catalogue
+    genome_data["result_directory"] = get_genome_result_path(genome_dir)
+    genome_data["biome"] = Biome.objects.filter(path=path).first()
+
+    genome_data = clean_genome_data(genome_data)
+
+    genome, _ = Genome.objects.update_or_create(
+        accession=accession, defaults=genome_data
+    )
+    genome.save()
+
+    logger.info(f"Uploaded genome and metadata for {accession}")
+
+    upload_cog_results(genome, genome_dir)
+    upload_kegg_class_results(genome, genome_dir)
+    upload_kegg_module_results(genome, genome_dir)
+    upload_antismash_geneclusters(genome, genome_dir)
+    upload_genome_files(genome, genome_dir, has_pangenome)
+
+    return accession
+
+
+def clean_genome_data(genome_data):
+    """
+    Clean genome data by removing unnecessary fields and ensuring required fields are present.
+    """
+    genome_data.pop("gold_biome", None)
+    if "annotations" not in genome_data:
+        genome_data["annotations"] = Genome.default_annotations()
+
+    geo_origin = genome_data.pop("geographic_origin", None)
+    if geo_origin:
+        genome_data["geo_origin"] = GeographicLocation.objects.get_or_create(
+            name=geo_origin
+        )[0]
+    genome_data.pop("genome_accession", None)
+    genome_data.pop("pangenome", None)
+
+    return genome_data
+
+
+def finalize_catalogue(catalogue):
+    catalogue.calculate_genome_count()
+    catalogue.save()
+
+
+@flow(name="import_genomes_flow")
+def import_genomes_flow(
```

Comment:
Are we at risk of having an invalid db connection? The import process could take some time, maybe we should refresh the db connection every now and then?

Reply:
@mberacochea I've added close_old_connections before genome db operations:

```python
   close_old_connections()
    genome, _ = Genome.objects.update_or_create(
        accession=accession, defaults=genome_data
    )
```

and also:

```python
    for genome_dir in genome_dirs:
        close_old_connections()
        genome_accession = process_genome_dir(catalogue, genome_dir)
        genome_accessions.append(genome_accession)
 ```